ncard_app/templates/predefinedQueries/phone_book.html

```diff
@@ -0,0 +1,97 @@
+{% extends 'events/base.html' %}
+{% load crispy_forms_tags %}
+
+{% block title %}Phone Book{% endblock %}
+
+{% block content%}
+    <h1>Phone Book</h1>
+
+        {% csrf_token %}
+    <style>
+        table, th, td {
+            border: 1px solid black;
+        }
+    </style>
+<table id="query-form" >
+    {% csrf_token %}
+    <tr>
+        <th> Title </th>
+        <th> Full Name </th>
+        <th> Organization Name </th>
+        <th> Phone(office) </th>
+        <th> Phone(Mobile) </th>
+        <th> Email </th>
+    </tr>
+    {% for ele in data %}
+        <tr>
+            <td>
+                {{ ele.primary_contact.title }}
+            </td>
+            <td>
+                {% if ele.primary_contact.surname_first %} {{ ele.primary_contact.surname }}
+                    {{ ele.primary_contact.given_name }}
+                {% else %}  {{ ele.primary_contact.given_name }}  {{ ele.primary_contact.surname }}
+                {% endif %}
+            </td>
+            <td>
+                {{ ele.name }}
+            </td>
+            <td>
+                {{ ele.primary_contact.contact.phone_office }}
+            </td>
+            <td>
+                {{ ele.primary_contact.contact.phone_mobile }}
+            </td>
+            <td>
+                {{ ele.primary_contact.contact.email }}
+            </td>
+        </tr>
+    {% endfor %}
+</table>
+<button onclick="get_export()">export</button>
+
+{% endblock %}
+
+
+{% block custom_js %}
+    <script>
+function get_export() {
+        const table = document.getElementById("query-form");
+        const data = []
+
+        for (var i = 0, rows = table.rows.length; i < rows; i++) {
+            for (var j = 0, cells = table.rows[i].cells.length; j < cells; j++) {
+                if (!data[i]) {
+                    data[i] = new Array();
+                }
+                data[i][j] = table.rows[i].cells[j].innerHTML
+            }
+        }
+        console.log(data)
+        const trans = JSON.stringify(data)
+        const csrfToken = $("[name='csrfmiddlewaretoken']").val()
+        params = {
+            'datalist': trans,
+
+            'csrfmiddlewaretoken': csrfToken,
+
+        }
+        $.ajaxSettings.async = false
+        $.post("{% url 'export' %}", params, function (data) {
+
+            var BOM = '\uFEFF';
+            csvString = BOM + data;
+            var a = document.createElement('a');
+            a.href = 'data:attachment/csv,' + encodeURI(csvString);
+            a.target = '_blank';
+            a.download = 'organisation.csv';//文件名
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
```

Comment:
Use this method to make a downloadable file in JS: https://stackoverflow.com/a/64908345

You don't need to add the 'a' element to the DOM for it to work.